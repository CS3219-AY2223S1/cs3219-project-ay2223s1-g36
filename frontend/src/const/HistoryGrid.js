import { formatDate } from '../utils/history';

export const qnHistoryCols = [
  { field: 'id', headerName: 'ID', flex: 1 },
  {
    field: 'ques_title',
    headerName: 'Question Title',
    flex: 3
  },
  { field: 'difficulty', headerName: 'Difficulty Level', flex: 2 },
  { field: 'language', headerName: 'Language', flex: 2 },
  {
    field: 'partner',
    headerName: 'Match',
    flex: 2,
    valueFormatter: (params) => {
      return params.value != null ? `@${params.value}` : '';
    }
  },
  {
    field: 'createdAt',
    headerName: 'Date of Attempt',
    flex: 3,
    valueFormatter: (params) => {
      return params.value != null ? `${formatDate(params.value)}` : '';
    }
  }
];
