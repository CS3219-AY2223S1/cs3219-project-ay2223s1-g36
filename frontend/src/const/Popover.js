export const MENU_OPTIONS = [
  {
    label: 'Home',
    icon: 'eva:home-fill',
    linkTo: '/'
  },
  {
    label: 'Profile',
    icon: 'eva:person-fill',
    linkTo: '#'
  },
  {
    label: 'Settings',
    icon: 'eva:settings-2-fill',
    linkTo: '#'
  }
];
