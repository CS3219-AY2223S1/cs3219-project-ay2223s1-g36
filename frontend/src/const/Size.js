export const DRAWER_WIDTH = 280;
export const APPBAR_MOBILE = 64;
export const APPBAR_DESKTOP = 92;
export const CHAT_DRAWER_WIDTH = 320;
