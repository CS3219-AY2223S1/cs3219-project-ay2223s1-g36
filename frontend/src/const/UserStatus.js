export const USER_STATUS_ONLINE = 'online';
export const USER_STATUS_DISCONNECTED = 'disconnected';
export const USER_STATUS_OFFLINE = 'offline';
