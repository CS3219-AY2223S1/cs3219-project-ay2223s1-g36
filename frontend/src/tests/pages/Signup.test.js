import renderer from 'react-test-renderer';
import SignupPage from '../../pages/Signup';
import { emptyFunction } from '../const';
import { RenderRouteWithOutletContext } from '../helpers/RenderRouteWithOutletContext';

const mockOutletContext = {
  username: 'test',
  setUsername: emptyFunction,
  password: '#TeSt123',
  setPassword: emptyFunction,
  handleSignup: emptyFunction
};

it('Signup page renders correctly', () => {
  const tree = renderer
    .create(
      <RenderRouteWithOutletContext context={mockOutletContext}>
        <SignupPage />
      </RenderRouteWithOutletContext>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
