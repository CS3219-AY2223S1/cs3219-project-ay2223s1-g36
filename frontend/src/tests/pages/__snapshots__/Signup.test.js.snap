// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup page renders correctly 1`] = `
Array [
  <h3
    className="MuiTypography-root MuiTypography-h3 css-1yzu38a-MuiTypography-root"
  >
    Sign Up
  </h3>,
  <div
    className="MuiFormControl-root MuiTextField-root css-o9dsj-MuiFormControl-root-MuiTextField-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={true}
      htmlFor=":r0:"
      id=":r0:-label"
    >
      Username
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={true}
        className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id=":r0:"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value="test"
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-1in441m"
        >
          <span>
            Username
          </span>
        </legend>
      </fieldset>
    </div>
  </div>,
  <div
    className="MuiFormControl-root MuiTextField-root css-1o50mq8-MuiFormControl-root-MuiTextField-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={true}
      htmlFor=":r1:"
      id=":r1:-label"
    >
      Password
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id=":r1:"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="password"
        value="#TeSt123"
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-1in441m"
        >
          <span>
            Password
          </span>
        </legend>
      </fieldset>
    </div>
  </div>,
  <div
    className="MuiBox-root css-n4rzf0"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium  css-sghohy-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Sign up
    </button>
  </div>,
  <p
    className="MuiTypography-root MuiTypography-body2 css-11huu28-MuiTypography-root"
  >
    Already have an account? 
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
      href="/welcome/login"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      Log in!
    </a>
  </p>,
]
`;
